### ISSUE TITLE
NVDA not reading Math in Microsoft Edge

### ISSUE THREAD

**Issue by bhavyashah** [NSoiffer/MathCAT#184]
I don't have much details to share on this other than I am using Microsoft Edge 117.0.2045.47, NVDA 2023.3beta3, and MathCAT version 0.3.6. NVDA simply doesn't read Math blocks sometimes until it starts reading them/I restart NVDA/I reopen the file. Sometimes, it won't read the Math for a while and other times, it fixes itself with a few Alt+Tabs out of and back into the document or just navigating around or giving it a minute. The only helpful info I have is the log snippet below:
IO - inputCore.executeGesture (22:01:34.891) - winInputHook (9544):
Input: kb(laptop):upArrow
DEBUGWARNING - NVDAObjects.IAccessible.ia2Web.Math._get_mathMl (22:01:34.921) - MainThread (9220):
Error retrieving math. Not supported in this browser or ISimpleDOM COM proxy not registered.
Traceback (most recent call last):
  File "NVDAObjects\IAccessible\ia2Web.pyc", line 334, in _get_mathMl
  File "comtypes\__init__.pyc", line 1198, in QueryInterface
  File "monkeyPatches\comtypesMonkeyPatches.pyc", line 32, in __call__
_ctypes.COMError: (-2147221164, 'Class not registered', (None, None, None, 0, None))
IO - speech.speech.speak (22:01:34.930) - MainThread (9220):
Speaking ['clickable']

Is there a particular browser that works better with NVDA in terms of Math?

**Comment by andrew-l-d** [comment]
I cannot replicate the issue.  Also using NVDA 2023.3 Beta 3.  Using Windows 10.
Andrew

**Comment by bhavyashah** [comment]
I want to be as helpful as I can with getting to the bottom of this
issue, because it is pretty ugly. I might have implied like the issue
sorts itself out given a couple minutes. That is not the case. Just
now, I was struggling to get NVDA to read the Math content beyond
"clickable", and this was in the speech style section (examples for
simple speak) of https://nsoiffer.github.io/MathCAT/users.html.
Restarting NVDA did not help. Switching to Chrome did not help. In
fact, installing Math Player and then restarting NVDA with add-ons
disabled did not help either. I installed Firefox and tried that, and
in my two attempts to load the page in Firefox, NVDA has been reading
Math on it; and once NVDA can read some Math somewhere, it can read
all Math anywhere, meaning viewing some Math in Firefox triggers NVDA
to begin reading Math in Microsoft Edge too. Hopefully this Firefox
trigger works consistently because I found no other work-around (other
than waiting for things to maybe sort themselves out).

On 10/1/23, andrew-l-d ***@***.***> wrote:
> I cannot replicate the issue.  Also using NVDA 2023.3 Beta 3.  Using Windows
> 10.
> Andrew
>
>
> --
> Reply to this email directly or view it on GitHub:
> https://github.com/NSoiffer/MathCAT/issues/184#issuecomment-1742454897
> You are receiving this because you authored the thread.
>
> Message ID: ***@***.***>


-- 
Kind Regards,
Bhavya Shah
B.S. in Mathematical and Computational Science | Stanford '24
LinkedIn: https://www.linkedin.com/in/bhavyashah125/

**Comment by andrew-l-d** [comment]
I usually use either Firefox or Chrome.  NVDA with Chrome read the maths on the Userguide page with no hiccoughs.  I then restarted NVDA and opened the page in Edge.  I still cannot replicate the issue.  Will try on another machine, not before tomorrow, and report if problems arise.
Andrew

**Comment by NSoiffer** [comment]
I sometimes see NVDA simply not read a page -- text or math. Typically moving the focus out of the window and back in solves the problem. But I've had cases where nothing I do works. This seemed to be independent of whether there was math on the page or not. Sometimes I've noticed a Windows update is pending and some other things are also flaky. A reboot has always solved the problem.

The log message about this low level functionality being missing is very suspicious. Have you tried rebooting?

**Comment by bhavyashah** [comment]
Well, I restarted my computer and haven't faced this issue in the day since. I shouldn't have missed that basic troubleshooting step. I'll reopen if this happens again.
